reactor-core/src/main/java/reactor/core/publisher/FluxBufferTimeout.java

```diff
@@ -320,6 +320,9 @@ public void request(long n) {
 		}
 
 		final void requestMore(long n) {
+			if (n < 1L) {
```

Comment:
Sounds like a bug somewhere in the core logic

Reply:
I'm not entirely sure... 
I think this scenario is valid and leads to the above state:
  - buffer size 10
  - let's assume a first request of 2 buffers
  - let's also assume the source is slow to emit at first and delivers 2 elements, then times out => first buffer request is met with timeout
  - then it emits one more element and stalls again => second buffer request is met with timeout also
  - `request(1)` comes in from downstream

At this point we have `outstanding` at `2x10 - 3` = `17`.
A request for one buffer translates to `requestLimit = 10`.
We have sufficient outstanding request to cover for that buffer and in `requestMore(n)` where `n = requestLimit - outstanding`, `n` is negative.

I think that this check should probably be better made in the `else` branch of `request` though, wdyt ?